src/mdio/segy/blocked_io.py

```diff
@@ -48,21 +57,21 @@ def to_zarr(segy_file: SegyFile, grid: Grid, data_array: Array, header_array: Ar
     chunker = ChunkIterator(data_array, chunk_samples=False)
     num_chunks = len(chunker)
 
-    # For Unix async writes with s3fs/fsspec & multiprocessing, use 'spawn' instead of default
-    # 'fork' to avoid deadlocks on cloud stores. Slower but necessary. Default on Windows.
-    num_cpus = int(os.getenv("MDIO__IMPORT__CPU_COUNT", default_cpus))
-    num_workers = min(num_chunks, num_cpus)
+    # Determine number of workers
+    num_cpus_env = int(os.getenv("MDIO__IMPORT__CPU_COUNT", default_cpus))
+    num_workers = min(num_chunks, num_cpus_env)
     context = mp.get_context("spawn")
-    executor = ProcessPoolExecutor(max_workers=num_workers, mp_context=context)
 
     # Chunksize here is for multiprocessing, not Zarr chunksize.
     pool_chunksize, extra = divmod(num_chunks, num_workers * 4)
-    pool_chunksize += 1 if extra else pool_chunksize
+    pool_chunksize += 1 if extra else 0
```

Comment:
this is interesting; what was the behaviour before?

Was it adding pool_chunksize if not extra?

Maybe simpler to

```python
if not extra:
    pool_chunksize += 1
```

wdyt